eclipse-sirius/sirius-web
# [3763] Implements the Selection Dialog TreeView

Bug: https://github.com/eclipse-sirius/sirius-web/issues/3763

# Pull request template

## General purpose
What is the main goal of this pull request?

## Project management

## Architectural decision records (ADR)

## Dependencies

## Frontend

This section is not relevant if your contribution does not come with changes to the frontend.

### General purpose

### Typing
We need to improve the typing of our code, as such, we require every contribution to come with proper TypeScript typing for both changes contributing new files and those modifying existing files.
Please ensure that the following statements are true for each file created or modified (this may require you to improve code outside of your contribution).

- Hooks are properly typed:

## Backend

This section is not relevant if your contribution does not come with changes to the backend.

### General purpose

### Architecture

## Review

### How to test this PR?

- import both :
[testSelectionDialogModel.zip](https://github.com/user-attachments/files/16694581/testSelectionDialogModel.zip)
[TestSelectionDialogStudio.zip](https://github.com/user-attachments/files/16694588/TestSelectionDialogStudio.zip)
- Open the diagram in testSelectionDialogModel
- Use the selection tool
- Select an existing element (only entity1 are selectable)
- Perform finish : the element should be renamed
